unittests/osr_test.py

```diff
@@ -0,0 +1,153 @@
+"""OSR testing of CEF Python."""
+
+import unittest
+import _test_runner
+from cefpython3 import cefpython as cef
+import platform
+import sys,time,os
+import base64
+
+g_datauri_data = """
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Title</title>
+    <script>
+        function print(msg) {
+            console.log(msg + " [JS]");
+        }
+        window.onload = function(){
+            document.addEventListener("mousedown", function( e ) {
+              print('mousedown:'+e.pageX+","+e.pageY)
+            }, false);
+            document.addEventListener("mouseup", function( e ) {
+              print('mouseup:'+e.pageX+","+e.pageY)
+            }, false);
+
+            document.addEventListener("mousemove", function( e ) {
+              print('mousemove:'+e.pageX+","+e.pageY)
+            }, false);
+
+            document.addEventListener("dragenter", function( e ) {
+              print('dragenter:'+e.pageX+","+e.pageY)
+            }, false);
+
+            document.addEventListener("dragover", function( e ) {
+              print('dragover:'+e.pageX+","+e.pageY)
+            }, false);
+            document.addEventListener("drop", function( e ) {
+              print('drop:'+e.pageX+","+e.pageY)
+            }, false);
+
+
+        }
+        print('window.onload init')
+    </script>
+    <style>
+        .box{
+            background-color: #F00;
+            height: 200px;
+        }
+    </style>
+</head>
+<body>
+
+<input id="selectFile" type="file">
+<div class="box"></div>
+</body>
+</html>
+
+"""
+
+g_datauri = "data:text/html;base64," + base64.b64encode(g_datauri_data.encode(
+    "utf-8", "replace")).decode("utf-8", "replace")
+VIEWPORT_SIZE = (1024, 768)
+
+g_subtests_ran = 0
+
+def subtest_message(message):
+    global g_subtests_ran
+    g_subtests_ran += 1
+    print(str(g_subtests_ran) + ". " + message)
+    sys.stdout.flush()
+
+class Handler(object):
+    def __init__(self):
+        self.OnPaint_called = False
+
+    def OnConsoleMessage(self, message, **_):
+        if "error" in message.lower() or "uncaught" in message.lower():
+            self.javascript_errors_False = True
+            raise Exception(message)
+        else:
+            # Check whether messages from javascript are coming
+            self.OnConsoleMessage_True = True
+            subtest_message(message)
+
+    def OnDragEnter(self, browser, dragData, mask):
+        subtest_message('cef.OnDragEnter() OK')
+        return False
+
+    def StartDragging(self,browser, drag_data, allowed_ops, x, y):
+        subtest_message('cef.StartDragging() OK')
+        return False
+
+    def OnLoadEnd(self,browser,frame,http_code):
+        subtest_message('cef.OnLoadEnd() OK')
+        #testing mouse event
+        browser.SendMouseClickEvent(300, 20, cef.MOUSEBUTTON_LEFT, False, 1, cef.EVENTFLAG_LEFT_MOUSE_BUTTON)
+        browser.SendMouseMoveEvent(305, 25, False, cef.EVENTFLAG_LEFT_MOUSE_BUTTON)
```

Comment:
Add the code below before send the first mouse click event. As I explained on the Forum, you must first send move event to the location at which you will trigger a click event. Try this:
```
browser.SendMouseMoveEvent(305, 25, False, 0)
browser.SendMouseClickEvent(300, 20, cef.MOUSEBUTTON_LEFT, False, 1, cef.EVENTFLAG_LEFT_MOUSE_BUTTON)
browser.SendMouseMoveEvent(305, 25, False, cef.EVENTFLAG_LEFT_MOUSE_BUTTON)
browser.SendMouseClickEvent(305, 25, cef.MOUSEBUTTON_LEFT, True, 1, cef.EVENTFLAG_LEFT_MOUSE_BUTTON)
```

If this code doesn't work as expected then add comment about it that StartDragging is currently never called and is on TODO list. Add a comment with a link to the Forum post in which I referenced code from upstream CEF unit tests on how it should be done.